docker-compose.yml

```diff
@@ -429,10 +415,117 @@ services:
       - 127.0.0.1:8093:8080
     networks:
       placeos:
-        ipv4_address: 172.31.231.19
     volumes:
       - type: volume
         source: rethink-data
         target: /data/rethinkdb_data
     environment:
       TZ: $TZ
+
+    # Logging
+
+  # Aggregates logs and forwards them to Elasticsearch.
+  logstash:
+    image: docker.elastic.co/logstash/logstash-oss:${ELASTIC_VERSION:-7.6}
+    restart: always
+    container_name: logstash
+    hostname: logstash
+    expose:
+      - 12201/udp
+      - 5044
+    ports:
+      - "127.0.0.1:12201:12201/udp"
+    networks:
+      placeos:
+    volumes:
+      - ./config/logstash/config:/config
+      - ./config/logstash/patterns:/opt/logstash/extra_patterns
+    restart: always
+    logging:
+      driver: json-file
+      options:
+        max-size: 99m
+    command: logstash -f /config
+
+  # Run 'docker-compose run --rm validate-logstash-config' to quickly check the logstash config.
+  validate-logstash-config:
+    container_name: validate-logstash
+    image: docker.elastic.co/logstash/logstash-oss:${ELASTIC_VERSION:-7.6}
+    volumes:
+      - ./config/logstash/config:/config
+    command: logstash -t -f /config
+
+  # Sends all container json-file logs to logstash
+  logspout:
+    image: vincit/logspout-gelf
+    hostname: $MONITOR_HOSTNAME
+    container_name: logspout
+    networks:
+      placeos:
+    volumes:
+      - /var/run/docker.sock:/var/run/docker.sock
+    command: gelf://$LOGSTASH_HOST:$LOGSTASH_PORT
+    restart: unless-stopped
+    logging:
+      driver: json-file
+      options:
+        max-size: 99m
+
+  # Gets metrics from host machine and send to elastic
+  metricbeat:
+    image: docker.elastic.co/beats/metricbeat-oss:${ELASTIC_VERSION:-7.6}
+    hostname: $MONITOR_HOSTNAME
+    container_name: metricbeat
+    user: root
+    networks:
+      placeos:
+    volumes:
+      - ${METRICBEAT_PROC_PATH:-/dev/null}:/hostfs/proc
+      - ${METRICBEAT_CGROUP_PATH:-/dev/null}:/hostfs/sys/fs/cgroup
+      - /:/hostfs:ro
+      - /var/run/docker.sock:/var/run/docker.sock:ro
+      - ./config/metricbeat.yml:/usr/share/metricbeat/metricbeat.yml
+    cap_add:
+      - SYS_PTRACE
+      - DAC_READ_SEARCH
+    command: ["metricbeat", "-e", "--strict.perms=false", "-system.hostfs=/hostfs", "-E", "output.elasticsearch.hosts=[$ELASTIC_HOST:$ELASTIC_PORT]"]
+    restart: unless-stopped
+    tty: true
+    logging:
+      driver: json-file
+      options:
+        max-size: 99m
```

Comment:
This is repeated, should it be replicated for other containers? 
At the least, might be worth putting into an anchor to reduce some duplication

Reply:
https://github.com/place-labs/partner-environment/pull/34/commits/86ddfaeea46ed8440d793b9173c9c92a2bf6d9d7